### ISSUE TITLE
Parameter with no value at all in Modelica.Blocks.Continuous.TransferFunction

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#374]
**Reported by svorkoetter on 4 Jun 2010 14:46 UTC**
In MSL 3.1, the block Modelica.Blocks.Continuous.TransferFunction contains this parameter declaration:
```text/x-modelica
parameter Real a[:] "Denominator coefficients of transfer function"
  annotation(Dialog(group="y = (2*s+3)/(4*s2+5*s+6)*u is defined as b={2,3}, a={4,5,6}"));
```

Notice that the parameter has no value _and_ no start value. In MSL 2.2.1, it did have a value of `{1,1}`.

Also, is that long ugly string really supposed to be displayed as the dialog box's group heading for this parameter??? Shouldn't that go into the string comment for the parameter instead? A better declaration for "a" (and likewise for "b") might be:

```text/x-modelica
parameter Real a[:] "Denominator coefficients of transfer function (e.g., 4*s2+5*s+6 is specified as {4,5,6})."
  annotation(Dialog(group="Numerator and denominator coefficients"));
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/374

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 4 Jun 2010 14:46 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 15 Jul 2010 16:21 UTC**
Yes, you are right, the value was changed from {1,1} to <nothing> when changing from version 2.2.2 to 3.0. I do not recall why this change was performed, because it is not backwards compatible. On the other hand the {1,1} value never makes sense ("a = s+1"). It might be useful that an instance of this block is usable, even if no parameter is provided. However, then a better default value would be "a = 1", because the default of the transfer function block would be then just "1". Would this be fine for you? If you agree, I will add the default 1 for a.

The "dialog(group)" annotation was introduced, because people might mix the "numerator" with the "denominator" term. So, if only a description string for the numerator and the denominator is present, such an error can still occur. If an example for a whole transfer function is given (as it is currently done with the group annotation), then errors from a user seem to be less likely.

In Dymola, this grouping displays nicely and the user of the block does not see that the definition is not so nice. So, in my opinion, the current solution is better.

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 15 Jul 2010 17:09 UTC**
Regarding the parameter value, the problem is that the lack of a declared value or start value means that it should inherit the default from type Real. The problem is that the dimension of a is not known without a declared value present, so it is impossible to flatten the component. I thought the whole idea of the use of "start=..." for parameters was introduced so that there would be a way to specify a default when there is no ideal default (just like R=1 was changed to R(start=1) for Resistors).

Regarding the group annotation, it does display "correctly" in the tools we've tried, but it does not display nicely. Having an equation as a group heading in a dialog box is very hacky looking. Of course there's nothing a tool can do about it, since it's up to the author of the code what the group heading should be. I find it unlikely that users will get "numerator" and "denominator" mixed up, especially since they are displayed one above the other, in the correct order, in the dialog box.

**Comment by modelica-trac-importer** [comment]
**Comment by stefanv on 15 Jul 2010 17:18 UTC**
Further comment. If the annotation is changed as I suggested, tools will display it like this:

```
Numerator and Denominator Coefficients -------------------------------------------------------------------------------------
b   [                     {1}]     Numerator coefficients of transfer function (e.g., 2*s+3 is specified as {2,3}).
a   [                   {1,1}]     Denominator coefficients of transfer function (e.g., 4*s2+5*s+6 is specified as {4,5,6}).
```

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 16 Jul 2010 06:06 UTC**
Fixed in 0955da482ae283237dd66c4fb26ee7765c982cf9

 * changed definition of a to "a=!!{1}" (in order to be symmetric,  "a" and "b" should either have a default value or a start value; for  convenience, the transfer function should have a value of !!{1} for b,  since the numerator is often 1. It is then best that the denominator  also gets this default value)
 * removed dialog group annotation as suggested and moved the  example to the parameter description texts.
